drivers/SmartThings/jbl/src/init.lua

```diff
@@ -0,0 +1,221 @@
+--  Copyright 2022 SmartThings
+--
+--  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+--  except in compliance with the License. You may obtain a copy of the License at:
+--
+--      http://www.apache.org/licenses/LICENSE-2.0
+--
+--  Unless required by applicable law or agreed to in writing, software distributed under the
+--  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+--  either express or implied. See the License for the specific language governing permissions
+--  and limitations under the License.
+--
+--  ===============================================================================================
+
+local log = require "log"
+
+local capabilities = require "st.capabilities"
+local Driver = require "st.driver"
+
+local discovery = require "discovery"
+local fields = require "fields"
+
+local jbl_discovery_helper = require "jbl.discovery_helper"
+local jbl_device_manager = require "jbl.device_manager"
+local jbl_capability_handler = require "jbl.capability_handler"
+
+local EventSource = require "lunchbox.sse.eventsource"
+
+local DEFAULT_MONITORING_INTERVAL = 300
+local CREDENTIAL_KEY_HEADER = "Authorization"
+
+local function handle_sse_event(driver, device, msg)
+  driver.device_manager.handle_sse_event(driver, device, msg.type, msg.data)
+end
+
+local function create_sse(driver, device, credential)
+  log.info("create_sse : dni = " .. tostring(device.device_network_id))
+  local conn_info = device:get_field(fields.CONN_INFO)
+
+  if not driver.device_manager.is_valid_connection(driver, device, conn_info) then
+    log.error("create_sse : invalid connection")
+    return
+  end
+
+  local sse_url = driver.device_manager.get_sse_url(driver, device, conn_info)
+  if not sse_url then
+    log.error("failed to get sse_url")
+  else
+    log.trace("Creating SSE EventSource for " .. device.device_network_id .. ", sse_url = " .. sse_url)
+    local eventsource = EventSource.new(sse_url, {[CREDENTIAL_KEY_HEADER] = credential}, nil)
+
+    eventsource.onmessage = function(msg)
+      if msg then
+        handle_sse_event(driver, device, msg)
+      end
+    end
+
+    eventsource.onerror = function(msg)
```

Comment:
nit: This `msg` will always be `nil` (the EventSource spec doesn't propagate an error message on unexpected close)

Reply:
@cchillerr This should be changed to a function with no arguments. EventSource errors don't have an error message unfortunately (as part of the spec)

Reply:
Fixed